internal/services/controllers/v1/task/controller.go

```diff
@@ -252,80 +253,113 @@ func (tc *TasksControllerImpl) Start() (func() error, error) {
 
 	ctx, cancel := context.WithCancel(context.Background())
 
+	spanContext, span := telemetry.NewSpan(ctx, "tasks-controller-start")
+
 	_, err = tc.s.NewJob(
 		gocron.DurationJob(tc.opsPoolPollInterval),
 		gocron.NewTask(
-			tc.runTenantTimeoutTasks(ctx),
+			tc.runTenantTimeoutTasks(spanContext),
 		),
 	)
 
 	if err != nil {
+		err := fmt.Errorf("could not schedule step run timeout: %w", err)
+
 		cancel()
-		return nil, fmt.Errorf("could not schedule step run timeout: %w", err)
+		span.RecordError(err)
+		span.End()
```

Comment:
I'm a bit surprised we need to call these two lines manually

Reply:
Generally you will do `defer span.End()` right after getting a new span inside a function but in this particular `Start()` function we are doing things little differently and depend on the `cleanup()` function to take care of such things.

As for `span.RecordErr()`, we do need to call it whenever we want to register errors and link it to the current span.